### DISCUSSION TITLE
NTSC C64 menu glitching

### DISCUSSION THREAD

**Discussion by hjf** [frntc/RAD#51] [General]
I built a RAD a few weeks ago and i had no problems with it so far other than some glitching (random characters here and there) while in menu mode. But REU mode worked fine.
But after some time it's just not working anymore.The glitching is so heavy it looks like snow on the screen, and if i press X to boot to BASIC with the REU, it either crashes or it shows a glitched "xx size reu"
I tried adjusting timings to no avail. it just makes the glitching worse. Could it be an issue with my machine instead? Maybe a PLA going bad?

I also have a NTSC flat C128, and I can't make my RAD work with it at all. Bad glitching but on some other discussions it seems it's "normal" for these not to work?

**Comment by insanity213** [comment]
Sounds like the one I had, it would mostly work on one of my many C64's but not on either NTSC or PAL 128.  It finally completely gave up the ghost but figured something I did was the cause.  I've got a couple more coming from another source today, hoping to get one RAD working reliably on at least 2 C64's then will give a go on a 128 again.

**Comment by hjf** [comment]
I figured it out. The problem was a "bad" 6510 CPU. Specifically, the PHI2 clock output. The output amplitude was wrong , it wasn't a nice 5Vpp output but rather it was offset a couple volts.
I suppose this throws off the RAD's clock detection.
There is a product called PHIagra that's supposed to buffer the output and take it to the right levels, but in my case it was fixed by swapping out the 6510.

**Reply by frntc** [reply]
Thanks for the info! Do you remember what voltage level Phi2 was? Was the slope also too shallow?

**Reply by hjf** [reply]
it was around 2v where it should have been 0 and maybe 3 or 4v (definitely not 5) where it should have been 5

the waveform was slow rise but fast fall. it had a "long" low time, but because of the slow rise, the high time was too short

![WhatsApp Image 2024-07-03 at 13 59 00](https://github.com/frntc/RAD/assets/603710/a08726a7-59e4-4786-b894-86dcf65caac4)

my first attempt was to add a 7414 Schmitt trigger inverter to square it up, but the output was always high. that's when I realized the low was too high so the Schmitt trigger output was always high.

my second attempt was to use a LM2903 comparator and a potentiometer to adjust the set point... but I was distracted and the computer was turned on. I had burned the 6510 phi2 output while soldering.

so I replaced the CPU and noticed the clock was similar to the bad ASCII art I posted but the zero was actually 0v. I tested rad and reu and they worked flawlessly

this problem was also affecting my 1541 ultimate.


I think an optional voltage comparator would be an useful addition to your design. I think this needs to be done in the analog domain as the voltage levels needed to trigger logic ICs are hardwired into them. with a comparator we could have an adjustment (and maybe a LED called "clock good" indicating that the waveform is a good 50% duty cycle.
